### PR TITLE
[FLINK-6930] [table] Forbid selecting window start/end on row-based Tumble/Slide windows

### PR DESCRIPTION
**Pull request by wuchong**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by hustfxj** [comment]
+1

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/logical/operators.scala [hunk 1]
```diff
@@ -652,6 +653,21 @@ case class WindowAggregate(
       case ValidationSuccess => // ok
     }
 
+    // validate property
+    if (propertyExpressions.nonEmpty) {
+      resolvedWindowAggregate.window match {
+        case TumblingGroupWindow(_, _, size) if isRowCountLiteral(size) =>
+          failValidation("Window start and Window end are not accessed " +
+                           "on row-counts Tumbling window.")
```
Comment (sunjincheng121) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/logical/operators.scala]: `row-counts` -> `row-count`

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/logical/operators.scala [hunk 2]
```diff
@@ -652,6 +653,21 @@ case class WindowAggregate(
...
+
+        case SlidingGroupWindow(_, _, size, _) if isRowCountLiteral(size) =>
+          failValidation("Window start and Window end are not accessed " +
+                           "on row-counts Sliding window.")
```
Comment (sunjincheng121) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/logical/operators.scala]: `row-counts` -> `row-count`

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/logical/operators.scala (hunk 1 above)
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/logical/operators.scala]: How about `"Window start and Window end cannot be selected for a row-count Tumbing window."`?

**Comment by fhueske** [comment]
Thanks for the fix @wuchong! 
I would adjust the error message as proposed but otherwise it's good to merge.

Thanks, Fabian

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/logical/operators.scala (hunk 1 above)
Comment (wuchong) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/logical/operators.scala]: Thank you @sunjincheng121 @fhueske  for the suggestion.

**Comment by wuchong** [comment]
Updated

**Comment by fhueske** [comment]
+1 to merge

**Comment by sunjincheng121** [comment]
+1 to merge

**Comment by wuchong** [comment]
merging...

**Pull request closed without merging**